service/application/src/test/java/com/solace/maas/ep/event/management/agent/applicationContextTests/SelfManagedAgentMessageHandlerBeansTests.java

```diff
@@ -0,0 +1,58 @@
+package com.solace.maas.ep.event.management.agent.applicationContextTests;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@ActiveProfiles("TEST")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = {
+        "eventPortal.gateway.messaging.standalone=false",
+        "eventPortal.managed=false"
+})
+class SelfManagedAgentMessageHandlerBeansTests {
```

Comment:
This test class makes sure we are loading the correct beans when ema is  in self managed agent mode